lib/hurley/client.rb

```diff
@@ -214,6 +193,41 @@ def body_receiver
       @body_receiver ||= [nil, BodyReceiver.new]
     end
 
+    def set_authorization_header_from_basic_auth!
+      header[:authorization] = url.basic_auth if url.basic_auth
```

Comment:
Nit picky, but I prefer this:

```ruby
if auth = url.basic_auth
  header[:authorization] = auth
end
```

This only runs `#basic_auth` once.  An alternative is adding a `#basic_auth?` method that checks for the existence of `@user`.